legislature/models.py

```diff
@@ -94,7 +64,44 @@ def votes_with_proposal(self):
         ).filter(representative=self)
 
 
-@receiver(post_save, sender=Representative)
-def create_memopolrepresentative_from_representative(instance, **kwargs):
-    memopol_representative = create_child_instance_from_parent(MemopolRepresentative, instance)
-    memopol_representative.save()
+def mempol_representative(sender, representative, data, **kwargs):
+    update = False
+    try:
+        memopol_representative = MemopolRepresentative.objects.get(
+            representative_ptr=representative)
+    except MemopolRepresentative.DoesNotExist:
+        memopol_representative = MemopolRepresentative(
+            representative_ptr=representative)
+
+        # Please forgive the horror your are about to witness, but this is
+        # really necessary. Django wants to update the parent model when we
+        # save a child model.
+        memopol_representative.__dict__.update(representative.__dict__)
+
+    try:
+        country = sorted(data.get('Constituencies', []),
+                         key=lambda c: c.get('end') if c is not None else 1
+                         )[-1]['country']
+    except IndexError:
```

Comment:
Really? <code>sorted</code> is supposed to return a python list iirc, so checking if the list it not empty then if the coontent (a dict?) has a key "country" shouldn't raise more queries. Am I reading this incorrectly?